[WIP]cnf950 - Provisioning and deploying a Distributed Unit (DU) (#28610)
openshift/openshift-docs: modules/cnf-deploying-the-du-infrastructure-profile.adoc

== Creating the Performance Addon Operator and DU performance profile

. Create and apply the performance profile, for example:
+
[source,yaml]
----
apiVersion: performance.openshift.io/v1
kind: PerformanceProfile
metadata:
# This profile is for typical lab DELL R640 with 2 x 26 pcores
  name: perf-example
spec:
  additionalKernelArgs:
  - nosmt
  cpu:
    # Temp. workaround for RT kernel bugs that consume too much
    # CPU power: add more isolated cores
    isolated: "1,3,5,7,9-51"  <1>
    reserved: "0,2,4,6,8"     <2>
  hugepages:
    defaultHugepagesSize: 1G
    pages:
    - count: 16   <3>
      size: 1G
      node: 0
  nodeSelector:
    # Pay attention to the node label, create MCP accordingly
    node-role.kubernetes.io/worker-cnf: ""
  numa:
    topologyPolicy: "restricted"
  realTimeKernel:
    # For CU should be false
    enabled: true
----
<1> Configure this line based upon the customer CPU hardware selected to run the DU.
<2> Configure this line based upon the customer CPU hardware selected to run the DU.
<3> Configure this line based upon the customer memory configuration selected to run the DU.

[id="cnf-creating-the-ptp-operator-and-slave-profile_{context}"]

> == Creating the PTP operator and slave profile

Reviewer: Operator

Author: Change made.

      ptp_dst_mac 01:1B:19:00:00:00
      p2p_dst_mac 01:80:C2:00:00:0E
      udp_ttl 1
      udp6_scope 0x0E
      uds_address /var/run/ptp4l
      #
      # Default interface options
      #
      clock_type OC
      network_transport UDPv4
      delay_mechanism E2E
      time_stamping hardware
      tsproc_mode filter
      delay_filter moving_median
      delay_filter_length 10
      egressLatency 0
      ingressLatency 0
      boundary_clock_jbod 0
      #
      # Clock description
      #
      productDescription ;;
      revisionData ;;
      manufacturerIdentity 00:00:00
      userDescription ;
      timeSource 0xA0
  recommend:
  - profile: "slave"
    priority: 4
    match:
    - nodeLabel: "ptp/slave"

EOF
----
<1> The interface selected needs to match the Linux interface name.
<2> `-2` configures Ethernet encapsulation of PTP.  `--summary_interval -4` sets the logging interval.
This is currently set to match `logSyncInterval -4`.
<3> `-n 24` must match the `domainNumber 24`.
<4> `domainNumber 24` must match the `-n 24`.
<5> These variables are set to enable the G.8275.1 profile for PTP.

[id="cnf-creating-the-sriov-operator-and-associated-profiles_{context}"]

> == Creating the SR-IOV operator and associated profiles

Reviewer: Operator

Author: Change made.

// CNF-950 4.7 Deploying the DU infrastructure profile
// Module included in the following assemblies:
//
// *cnf-provisioning-and-deploying-a-distributed-unit.adoc

[id="scalability_and_performance/cnf-deploying-the-du-infrastructure-profile_{context}"]
= Deploying the DU infrastructure profile

[id="cnf-creating-the-performance-addon-operator-and-du-performance-profile_{context}"]
== Creating the Performance Addon Operator and DU performance profile

> . Create the `kustomization.yaml` file, for example:
> +
> [source,yaml]
> ----
> apiVersion: kustomize.config.k8s.io/v1beta1
> kind: Kustomization
> 
> resources:
>   - performance-profile.yaml
> ----

Reviewer 2: ```suggestion

```

Author: Change made.

> . Create the `performance-profile.yaml` file, for example:

Reviewer 2: ```suggestion
. Create and apply the performance profile, for example:
```

Author: Change made.

Author: @vitus133 In the code bloc for this step, do you want to add the cat **<<EOF |  oc apply -f -**  & **EOF** statements?

+
[source,yaml]
----

> apiVersion: performance.openshift.io/v1

Reviewer 2: ```suggestion
cat <<EOF | oc apply -f -
apiVersion: performance.openshift.io/v1
```

> ----
> apiVersion: performance.openshift.io/v1
> kind: PerformanceProfile
> metadata:
> # This profile is for typical lab DELL R640 with 2 x 26 pcores
>   name: perf-example
> spec:
>   additionalKernelArgs:
>   - nosmt
>   cpu:
>     # Temp. workaround for RT kernel bugs that consume too much
>     # CPU power: add more isolated cores
>     isolated: "1,3,5,7,9-51"  <1>
>     reserved: "0,2,4,6,8"     <2>
>   hugepages:
>     defaultHugepagesSize: 1G
>     pages:
>     - count: 16   <3>
>       size: 1G
>       node: 0
>   nodeSelector:
>     # Pay attention to the node label, create MCP accordingly
>     node-role.kubernetes.io/worker-cnf: ""
>   numa:
>     topologyPolicy: "restricted"
>   realTimeKernel:
>     # For CU should be false
>     enabled: true
> ----

Reviewer 2: ```suggestion
cat <<EOF | oc apply -f -
apiVersion: performance.openshift.io/v1
kind: PerformanceProfile
metadata:
# This profile is for typical lab DELL R640 with 2 x 26 pcores
  name: perf-example
spec:
  additionalKernelArgs:
  - nosmt
  cpu:
    # Temp. workaround for RT kernel bugs that consume too much
    # CPU power: add more isolated cores
    isolated: "1,3,5,7,9-51"  <1>
    reserved: "0,2,4,6,8"     <2>
  hugepages:
    defaultHugepagesSize: 1G
    pages:
    - count: 16   <3>
      size: 1G
      node: 0
  nodeSelector:
    # Pay attention to the node label, create MCP accordingly
    node-role.kubernetes.io/worker-cnf: ""
  numa:
    topologyPolicy: "restricted"
  realTimeKernel:
    # For CU should be false
    enabled: true
EOF
```

> ----

Reviewer 2: ```suggestion
EOF
----
```

<1> Configure this line based upon the customer CPU hardware selected to run the DU.
<2> Configure this line based upon the customer CPU hardware selected to run the DU.
<3> Configure this line based upon the customer memory configuration selected to run the DU.

[id="cnf-creating-the-ptp-operator-and-slave-profile_{context}"]
== Creating the PTP operator and slave profile

> 
> . Create the `kustomization.yaml` for the `ptpconfig-slave.yaml` file, for example:
> +
> [source,yaml]
> ----
> apiVersion: kustomize.config.k8s.io/v1beta1
> kind: Kustomization
> 
> resources:
>   - ptpconfig-slave.yaml
> ----
> 

Reviewer 2: ```suggestion
```

Author: Change made.

> . Create the `ptpconfig-slave.yaml` file, for example:
> +

Reviewer 2: ```suggestion
. Create an apply the PTP configuration, for example:
+
```

Author: Change made.

[source,yaml]

> ----

Reviewer 2: ```suggestion
cat <<EOF |  oc apply -f -
```

Author: Change made.

      pi_integral_const 0.0
      pi_proportional_scale 0.0
      pi_proportional_exponent -0.3
      pi_proportional_norm_max 0.7
      pi_integral_scale 0.0
      pi_integral_exponent 0.4
      pi_integral_norm_max 0.3
      step_threshold 0.0
      first_step_threshold 0.00002
      max_frequency 900000000
      clock_servo pi
      sanity_freq_limit 200000000
      ntpshm_segment 0
      #
      # Transport options
      #
      transportSpecific 0x0
      ptp_dst_mac 01:1B:19:00:00:00
      p2p_dst_mac 01:80:C2:00:00:0E
      udp_ttl 1
      udp6_scope 0x0E
      uds_address /var/run/ptp4l
      #
      # Default interface options
      #
      clock_type OC
      network_transport UDPv4
      delay_mechanism E2E
      time_stamping hardware
      tsproc_mode filter
      delay_filter moving_median
      delay_filter_length 10
      egressLatency 0
      ingressLatency 0
      boundary_clock_jbod 0
      #
      # Clock description
      #
      productDescription ;;
      revisionData ;;
      manufacturerIdentity 00:00:00
      userDescription ;
      timeSource 0xA0
  recommend:
  - profile: "slave"
    priority: 4
    match:
    - nodeLabel: "ptp/slave"

> ----

Reviewer 2: ```suggestion
EOF
```

Author: Change made

<1> The interface selected needs to match the Linux interface name.
<2> `-2` configures Ethernet encapsulation of PTP.  `--summary_interval -4` sets the logging interval.
This is currently set to match `logSyncInterval -4`.
<3> `-n 24` must match the `domainNumber 24`.
<4> `domainNumber 24` must match the `-n 24`.
<5> These variables are set to enable the G.8275.1 profile for PTP.

[id="cnf-creating-the-sriov-operator-and-associated-profiles_{context}"]
== Creating the SR-IOV operator and associated profiles

> 
> . Create  the `kustomization.yaml` for the `sriov-network-node-policy-dpdk-mh.yaml` and `sriov-network-mh.yaml` files, for example:
> +
> [source,yaml]
> ----
> apiVersion: kustomize.config.k8s.io/v1beta1
> kind: Kustomization
> 
> resources:
>   - sriov-network-node-policy-dpdk-mh.yaml
>   - sriov-network-mh.yaml
> ----
> 

Reviewer 2: ```suggestion

```

Author: Change made.

> . Create the `sriov-network-node-policy-dpdk-mh.yaml` file, for example:

Reviewer 2: ```suggestion
. Create and apply the SR-IOV network node policy, for example:
```

Author: Change made.

+
[source,yaml]

> ----

Reviewer 2: ```suggestion
cat <<EOF |  oc apply -f -
```

Author: Change made.

apiVersion: sriovnetwork.openshift.io/v1
kind: SriovNetworkNodePolicy
metadata:
  name: policy-mh-dpdk-site-1-fqdn-worker1
  namespace: openshift-sriov-network-operator
spec:
# This works for Intel based NICs. <1>
# For Mellanox please change to:
#     deviceType: netdevice
#     isRdma: true
  deviceType: vfio-pci
  isRdma: false
  nicSelector:
# The exact physical function name must match the hardware used
    pfNames: ["ens1f1"] <2>
  nodeSelector:
    node-role.kubernetes.io/worker-cnf: ""
    feature.node.kubernetes.io/network-sriov.capable: "true"
  numVfs: 4
  priority: 10
  resourceName: mh_u_site_1_fqdn_worker1

> ----

Reviewer 2: ```suggestion
EOF
```

Author: Change made.

> <1> This file works for Intel and must change for Mellanox.

Reviewer 2: ```suggestion
<1> This file works for Intel and must change for Mellanox, as described in SR-IOV configuration notes chapter.
```
(please link to the chapter in  modules/cnf-provisioning-deploying-a-distributed-unit-(du)-manually.adoc)

Author: We can't use internal xrefs (our current build machine won't render them). I'll have to use hard text.

<2> Must be updated with the specific device on the server.

> . Create the `sriov-network-mh.yaml` file, for example:

Reviewer 2: ```suggestion
. Create the SR-IOV network, for example:
```

Author: Change made.

+
[source,yaml]

> ----

Reviewer 2: ```suggestion
cat <<EOF | ${OC_TOOL} apply -f -
```

Author: Change made.

apiVersion: sriovnetwork.openshift.io/v1
kind: SriovNetwork
metadata:
  name: mh-net-u-site-1-fqdn-worker1
  namespace: openshift-sriov-network-operator
spec:
  ipam:  |
    {
    }
  networkNamespace: mh-net-ns-site-1-fqdn-worker1
  resourceName: mh_u_site_1_fqdn_worker1
  vlan: 100  <1>
---
apiVersion: v1
kind: Namespace
metadata:
    name: mh-net-ns-site-1-fqdn-worker1

> ----

Reviewer 2: ```suggestion
EOF
```

Author: Change made.
